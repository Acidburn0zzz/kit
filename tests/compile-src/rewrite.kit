include "stdio.h";

rules MyRewriteRules {
    (5) => 6;
    (${x: Int} %% 1) => $x + 1;
    (apple) => true;
    (banana) => strawberry;
}

function main() {
    printf("%li\n", 5_i32);

    using rules MyRewriteRules {
        // custom operator; must be rewritten or will fail to compile
        var x: Int = 5;
        // var x = 5_i32;
        printf("%li\n", x);
        printf("%li\n", x %% 1);

        // var doesn't exist; must be rewritten or will fail to compile
        if (apple) {
            printf("apple => true\n");
        }

        var strawberry = true;
        if (banana) {
            printf("banana => strawberry\n");
        }
    }
}
