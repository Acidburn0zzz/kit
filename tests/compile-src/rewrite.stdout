5
6
7
apple => true
banana => strawberry
